### PR TITLE
[#1384] Add more types suggested replies

### LINKED ISSUE #1384: Support more types for Suggested Replies
**Current state** 
The typing of Suggested Replies only marks text as supported type 

```
export interface SuggestedReply {
  content: {
    text: string;
  };
}
```

**To Do** 
Suggested Replies need to support more types

### PR DESCRIPTION
**Pull request by thorstenairy**
closes #1384

### CODE REVIEW

File: lib/typescript/httpclient/model/Message.ts [hunk 1]
```diff
@@ -35,7 +35,7 @@ export interface Message extends Content {
   metadata?: MessageMetadata;
 }
 
-export interface SuggestedReply {
+export interface SuggestedReply extends Content {
```
Comment (AudreyKj) [inline lib/typescript/httpclient/model/Message.ts:38]: would this type work with the Content type? 

`export interface Content {
  id: string;
  content: any;
}`

File: frontend/ui/src/pages/Inbox/MessageInput/index.tsx [hunk 2]
```diff
@@ -71,13 +75,20 @@ const MessageInput = (props: MessageInputProps & ConnectedProps<typeof connector
...
+    if (selectedSuggestedReply) {
+      setSelectedSuggestedReply(null);
+      sendMessages({conversationId: conversation.id, message: selectedSuggestedReply.message.content}).then(() =>
+        setInput('')
+      );
       return;
     }
 
-    props.sendMessages(getTextMessagePayload(source, conversation.id, input)).then(() => setInput(''));
+    sendMessages(getTextMessagePayload(source, conversation.id, input)).then(() => setInput(''));
```
Comment (AitorAlgorta) [inline frontend/ui/src/pages/Inbox/MessageInput/index.tsx:91]: I think we could reduce some code in this method. Just use `sendMessages` once at the end, so the dynamic thing is the payload

File: frontend/ui/src/pages/Inbox/MessageInput/index.module.scss [hunk 3]
```diff
@@ -233,14 +241,14 @@
   z-index: 1;
 }
 
-.removeTemplateButton svg {
+.removeButton svg {
   width: 8px;
   height: 8px;
   margin-left: -1px;
```
Comment (AitorAlgorta) [inline frontend/ui/src/pages/Inbox/MessageInput/index.module.scss:247]: is this necessary?

File: frontend/ui/src/pages/Inbox/MessageInput/index.tsx [hunk 4]
```diff
@@ -139,17 +150,16 @@ const MessageInput = (props: MessageInputProps & ConnectedProps<typeof connector
       const jsonTemplate = template.content;
 
       if (
-        jsonTemplate.message.text &&
-        !jsonTemplate.message.suggestions &&
-        !jsonTemplate.message.quick_replies &&
-        !jsonTemplate.message.containsRichText &&
-        !jsonTemplate.message.attachments &&
-        !jsonTemplate.message.attachment
+        jsonTemplate.text &&
```
Comment (AitorAlgorta) [inline frontend/ui/src/pages/Inbox/MessageInput/index.tsx:153]: I think that at some point we will need helper functions like `isTextMessage` or `isRichCardMessage`

File: frontend/ui/src/pages/Inbox/SuggestedReplySelector/index.module.scss [hunk 5]
```diff
@@ -33,7 +33,7 @@
   width: max-content;
   height: auto;
   flex-shrink: 0;
-  max-width: 320px;
+  max-width: 400px;
```
Comment (AitorAlgorta) [inline frontend/ui/src/pages/Inbox/SuggestedReplySelector/index.module.scss:36]: I thought we were removing this, what about the carousels?

File: lib/typescript/httpclient/model/Message.ts [hunk 6]
```diff
@@ -35,7 +35,7 @@ export interface Message extends Content {
...
   content: {
     text: string;
```
Comment (AitorAlgorta) [inline lib/typescript/httpclient/model/Message.ts:40]: Not sure if this is right. `SuggestedReply` can have any type of message inside, not just text.

**Review by AitorAlgorta — COMMENTED** [review]
Left some comments & questions

File: frontend/ui/src/pages/Inbox/MessageInput/index.tsx (hunk 4 above)
Comment (lucapette) [reply frontend/ui/src/pages/Inbox/MessageInput/index.tsx:153]: I'd argue we're already at that point :D

File: frontend/ui/src/pages/Inbox/SuggestedReplySelector/index.module.scss (hunk 5 above)
Comment (thorstenairy) [reply frontend/ui/src/pages/Inbox/SuggestedReplySelector/index.module.scss:36]: works smoothly on big screens, if you use it with 13' the experience is broken without max-width. @AitorAlgorta

**Review by AitorAlgorta — DISMISSED** [review]
Small comments but LGTM

**Review by lucapette — APPROVED** [review]

**Pull request merged**